workflows/smrnaseq.nf

```diff
@@ -0,0 +1,241 @@
+
+    //
+    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+    //
+    INPUT_CHECK (
+        ch_input
+    ).map {
+        meta, fastq ->
+            meta.id = meta.id.split('_')[0..-2].join('_')
+            [ meta, fastq ] }
+    .dump(tag: 'map')
+    .groupTuple(by: [0])
+    .dump(tag: 'group')
+    .branch {
+        meta, fastq ->
+            single  : fastq.size() == 1
+                return [ meta, fastq.flatten() ]
+            multiple: fastq.size() > 1
+                return [ meta, fastq.flatten() ]
+    }
+    .set { ch_fastq }
+
+    //
+    // MODULE: Concatenate FastQ files from same sample if required
+    //
+    CAT_FASTQ (
+        ch_fastq.multiple
+    )
+    CAT_FASTQ.out.reads
+    .mix(ch_fastq.single)
+    .set { ch_cat_fastq }
+
+    //
+    // SUBWORKFLOW: mirtrace QC
+    //
+    MIRTRACE (ch_cat_fastq)
+    ch_software_versions = ch_software_versions.mix(MIRTRACE.out.versions.ifEmpty(null))
+
+
+    //
+    // SUBWORKFLOW: Read QC, extract UMI and trim adapters
+    //
+    FASTQC_TRIMGALORE (
+        ch_cat_fastq,
+        params.skip_fastqc || params.skip_qc,
+        params.skip_trimming
+    )
+    ch_software_versions = ch_software_versions.mix(FASTQC_TRIMGALORE.out.fastqc_versions.first().ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(FASTQC_TRIMGALORE.out.trimgalore_versions.first().ifEmpty(null))
+
+    reads_for_mirna = FASTQC_TRIMGALORE.out.reads
+    MIRNA_QUANT (
+        reference_mature,
+        reference_hairpin,
+        mirna_gtf,
+        reads_for_mirna
+    )
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.bowtie_versions.ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.samtools_versions.first().ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.seqcluster_versions.first().ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.mirtop_versions.ifEmpty(null))
+    ch_software_versions = ch_software_versions.mix(MIRNA_QUANT.out.merge_versions.ifEmpty(null))
+
+    //
+    // GENOME
+    //
+    genome_stats = Channel.empty()
+    if (fasta){
+        fasta_ch = file(fasta)
+        GENOME_QUANT ( fasta_ch, bt_index, MIRNA_QUANT.out.unmapped )
+        GENOME_QUANT.out.stats
+            .set { genome_stats }
+        MIRDEEP2 (FASTQC_TRIMGALORE.out.reads, GENOME_QUANT.out.fasta , GENOME_QUANT.out.indices, MIRNA_QUANT.out.fasta_hairpin, MIRNA_QUANT.out.fasta_mature)
+        ch_software_versions = ch_software_versions.mix(MIRDEEP2.out.versions.first().ifEmpty(null))
+
+    }
+
+    //
+    // MODULE: Pipeline reporting
+    //
+    ch_software_versions
+        .map { it -> if (it) [ it.baseName, it ] }
+        .groupTuple()
+        .map { it[1][0] }
+        .flatten()
+        .collect()
+        .set { ch_software_versions }
+
+    GET_SOFTWARE_VERSIONS (
+        ch_software_versions.map { it }.collect()
+    )
```

Comment:
Change this with the new implementation, once all the versions are reported by modules and subworkflow, see [here](https://github.com/nf-core/rnaseq/blob/964425e3fd8bfc3dc7bce43279a98d17a874d3f7/workflows/rnaseq.nf#L755-L757)
```suggestion
```